src/mongocxx/options/find.hpp

```diff
@@ -455,6 +455,16 @@ class MONGOCXX_API find {
     ///
     const stdx::optional<bsoncxx::document::view_or_value>& sort() const;
 
+    ///
+    /// Modifies the object to convert all modifiers into their respective top-level options. For
+    /// internal use only.
+    ///
+    /// @throws
+    ///   mongocxx::exception if any modifiers are invalid, or if the unsupported "$query" or
+    ///   "$explain" modifiers are set.
+    ///
+    MONGOCXX_PRIVATE find& convert_all_modifiers();
```

Comment:
Setting aside issues of ownership, I think it should return a copy.  We're passing options via `const`, so find() can't call this to mutate the options – it needs a clean copy anyway.